### DISCUSSION TITLE
PyTest-BDD and SeleniumBase

### DISCUSSION THREAD

**Discussion by kennedylunar** [seleniumbase/SeleniumBase#811] [Q&A] [answered]
Hello,

I am trying to do the SeleniumBase together with pytest-bdd, but when i do something like this:
      tests/
              ├── features/
               │   ├── search.feature
              ├── step_defs/
               │   ├── test_search.py

I tried running a single step 'Given' on the console it is running 2 test even though it is a single step but it is running. Then I do a 'Given' and 'When' then it could not detect the bdd format. 

Another thing is if I call sb in 2 different methods those it mean that I am running 2 different tests already?

          def test_ddg_home(sb):
              sb.open("https://www.duckduckgo.com")
          
          def test_pageTitle(sb):
              sb.assert_title("DuckDuckGo — Privacy, simplified.")

**Comment by mdmintz** [comment, marked as answer]
@kennedylunar 
1. Not sure about pytest-bdd; that's a completely different library.
2. That's two different tests you have there, and the second one will fail because only the first test navigates to the page that you want. However, if you run tests with the ``--reuse-session`` / ``--rs`` arg, it will keep the same browser window open for all tests, so your code would work in that specific case. Keep in mind, the tests will run in alphabetical order, unless you use pytest-ordering to select the exact order of tests.
